### DISCUSSION TITLE
Mühendislerin işleri

### DISCUSSION THREAD

**Discussion by Aliihsan91** [sonsuzus/sonsuzus#472] [Soru Cevap]
Her test için zaman sınırı: 2 saniye
Her test için hafıza sınırı: 512 megabayt

Bir yazılım ekibinde 1'den n'e kadar numaralandırılmış n mühendis bulunmaktadır. Patron, ek bir projeyi tamamlama koşuluyla ekibe bir prim sözü verir. Projenin toplamda k birimlik bir iş gereksinimi vardır. i'inci mühendise vaat edilen prim miktarı aᵢ burledir. Patron, belirli görevleri mühendislere atamaz; her mühendis projeye gönüllü olarak belirli miktarda (tam sayı olarak) iş birimi tamamlaması beklenir. Prim yalnızca proje tamamlanırsa ödenir, yani ekibin toplam gönüllü iş miktarı k'yi karşılarsa veya aşarsa prim ödenir.

Mühendislerin tamamlayabileceği iş miktarı sınırlandırılmamıştır. Ancak, her mühendis emeğini değerli görür. i'inci mühendis, bir birim işin maliyetini bᵢ burle olarak tahmin eder.

**Comment by Aliihsan91** [comment]
```import sys

input = sys.stdin.read
data = input().split()

n, k = int(data[0]), int(data[1])  # Mühendis sayısı ve toplam iş gereksinimi
a = list(map(int, data[2:n+2]))  # Prim miktarları
b = list(map(int, data[n+2:2*n+2]))  # Birim iş maliyetleri

engineers = sorted(
    [(i, a[i], b[i], a[i] / b[i]) for i in range(n)], 
    key=lambda x: x[3], 
    reverse=True
)

work = [0] * n
remaining_work = k

for idx, ai, bi, _ in engineers:
    if remaining_work <= 0:  # Eğer gereken iş tamamlandıysa döngüden çık
        break
    max_work = ai // bi  # Faydanın negatif olmaması için maksimum iş miktarı
    contribution = min(max_work, remaining_work)  # Mühendis yapabileceği kadar çalışır
    work[idx] = contribution  # Mühendisin yaptığı iş miktarını kaydet
    remaining_work -= contribution  # Kalan işi güncelle

print(" ".join(map(str, work)))
```

**Reply by sonsuzus** [reply]
çok chatgpt olmuş. ayrıca soruda örnek girdi ve çıktılar eksik. soru tam anlaşılmıyor. cevap hiç anlaşılmıyor ve yanlışlar var.
